File: TypeScript/codewars/Growth-of-a-Population.ts
export const nbYear = (p0: number, percent: number, aug: number, p: number): number => {
	let years = 0;
	let newPopulation = p0;

	while (newPopulation < p) {
		years++;
		newPopulation = calculatePopulation(newPopulation, percent, aug);
	}

	return years;
};

function calculatePopulation(p0: number, percent: number, aug: number) {
	return Math.floor(p0 + (p0 * percent) / 100 + aug);
}

File: Csharp/CodeWars/src/Resolutions/GrowthOfAPopulation.cs
namespace CodeWars.Resolutions;

public class GrowthOfAPopulation
{
    public static int NbYear(int p0, double percent, int aug, int p)
    {
        int years = 0;
        int newPopulation = p0;

        while (newPopulation < p)
        {
            years++;
            newPopulation = CalculatePopulation(newPopulation, percent, aug);
        }

        return years;
    }
    private static int CalculatePopulation(int p0, double percent, int aug) =>
        (int)Math.Floor(p0 + p0 * percent / 100 + aug);
}
